app/src/main/java/org/simple/clinic/util/ErrorResolver.kt

```diff
@@ -60,11 +60,11 @@ object ErrorResolver {
 
 sealed class ResolvedError(val actualCause: Throwable) {
 
-  class NetworkRelated(actualCause: Throwable) : ResolvedError(actualCause)
+  data class NetworkRelated(private val _actualCause: Throwable) : ResolvedError(_actualCause)
```

Comment:
Just for my information, what is the reason to add `_` to `actualCause`?
Is this a kind of best practice?

Reply:
The only reason is because `ResolvedError` also exposes a property and I cannot make these types data class unless I convert them to properties. This `_` was added to avoid the name clash.

The "right" solution, IMO, is maybe to not expose the property from `ResolvedError` and let callers get the cause from the individual error types but that is too invasive a change to do at the moment.

I could also rename them `causeInternal` to fit with our usual convention.

Reply:
Discussed offline: We will retain the names as they are for now.